### PR TITLE
Precommit checks for license

### PR DESCRIPTION
**Pull request by superm1**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: contrib/ci/check-license.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    for fn in glob.glob("**/*.[c|h|py|sh]", recursive=True):
+        if "meson-private" in fn:
+            continue
+        license = __get_license(fn)
+        if not license:
+            print("{} does not specify a license".format(fn))
+            rc = 1
+            continue
+        if not "GPL" in license:
+            print("{} does not contain LGPL or GPL ({})".format(fn, license))
```
Comment (hughsie) [inline contrib/ci/check-license.py:35]: Are we not LGPL2+ everywhere?

File: plugins/flashrom/fu-plugin-flashrom.c [hunk 2]
```diff
@@ -3,21 +3,7 @@
...
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with this program; if not, write to the Free Software
- * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ * SPDX-License-Identifier: GPL-2+
```
Comment (hughsie) [inline plugins/flashrom/fu-plugin-flashrom.c:6]: I think this should be LGPLv2, no?

Comment (superm1) [reply plugins/flashrom/fu-plugin-flashrom.c:6]: You wrote it originally, so I guess that's actually our intent..

File: contrib/ci/check-license.py (hunk 1 above)
Comment (superm1) [reply contrib/ci/check-license.py:35]: Here are the exceptions:
1. EMMC plugin was leveraged from GPL code, I don't think we can just reclassify it.
2. BCM57xx is GPL-2+
3. CPU plugin is GPL-2+
4. Intel SPI plugin is GPL-2+
5. Flashrom is GPL-2

**Review by hughsie — APPROVED** [review]

**Pull request merged**